### ISSUE TITLE
Incorrect long text visibility when responsive is stacked or stackedFullWidth

### ISSUE THREAD

**Issue by ramosmarco** [gregnb/mui-datatables#1188]
Hello everyone, When the responsive option is set to stacked or stackedFullWidth, lonk text cells are not showed correctly, here is what I found and how I solved.

1) When stacked is used, the long text is missing and there is no way to see it.
2) For stackedFullWidth is possible to see it but it overlap the table.

## Current Behavior

This is how it looks with sort text

stacked
![stacked_ok](https://user-images.githubusercontent.com/46763360/73974856-f7c1d580-48d9-11ea-97e5-b9c1845d4ff2.jpg)

stackedFullWidth
![stackedFullWidth_ok](https://user-images.githubusercontent.com/46763360/73975479-3f952c80-48db-11ea-85fb-20d900aa55fd.jpg)

This is how it looks with long text

stacked, as you can see part of the text is not visible
![stacked_wrong](https://user-images.githubusercontent.com/46763360/73975589-8125d780-48db-11ea-8810-004df06ffbba.jpg)

stackedFullWidth, here all the text is showed but it overlap the table
![stackedFullWidth_wrong](https://user-images.githubusercontent.com/46763360/73975633-9864c500-48db-11ea-9194-8679b1f07ce9.jpg)

## Solution
1) Add a father MuiThemeProvider component 
2) Overwrite the style on MUIDataTableBodyCell / stackedCommon / '@media (max-width:959.95px)' change height to 100%, whiteSpace to normal and display to inherit (this last only needed for stackedFullWidth), I changed other styles but just better looks like make title bold and remove underline)

Here is the code changed
```
import React from "react";
import ReactDOM from "react-dom";
import MUIDataTable from "mui-datatables";
import { createMuiTheme, MuiThemeProvider } from '@material-ui/core/styles';

const getMuiTheme = () => createMuiTheme({
  overrides: {
      MuiTableCell: {
          root: {
              '@media (max-width:959.95px)': { 
                  borderBottom: "none",
                  padding: "1px",
              },
          }
      },
      MUIDataTableBodyCell: {
          stackedCommon: {
              '@media (max-width:959.95px)': { 
                  whiteSpace: "normal",
                  height: "100%",
                  paddingLeft: "5px",
                  display: "inherit"
              },
          },
          cellHide:{
              fontWeight: "bold",
              paddingTop: "15px",
          }
      },
      MuiTableRow: {
          root: {
              borderBottom: "1px solid rgba(224, 224, 224, 1)",
          }
      }
  }
});

class App extends React.Component {
  render() {
    const columns = ["Name", "Title", "Location", "Age", "Salary"];
    const data = [
      ["Gabby George","Business Analyst, very long text, for a very importa person, the big boss","Minneapolis",30,"$100,000"],
      ["Aiden Lloyd", "Business Consultant", "Dallas", 55, "$200,000"],
      ["Jaden Collins", "Attorney", "Santa Ana", 27, "$500,000"],
      ["Franky Rees", "Business Analyst", "St. Petersburg", 22, "$50,000"],
      ["Aaren Rose", "Business Consultant", "Toledo", 28, "$75,000"],
      ["Blake Duncan","Business Management Analyst","San Diego",65,"$94,000" ],
      ["Frankie Parry", "Agency Legal Counsel", "Jacksonville", 71, "$210,000"],
      ["Lane Wilson", "Commercial Specialist", "Omaha", 19, "$65,000"],
      ["Robin Duncan", "Business Analyst", "Los Angeles", 20, "$77,000"],
      ["Mel Brooks", "Business Consultant", "Oklahoma City", 37, "$135,000"],
      ["Harper White", "Attorney", "Pittsburgh", 52, "$420,000"]
    ];

    const options = {
      responsive: "stackedFullWidth", 
    };

    return (
      <MuiThemeProvider theme={getMuiTheme()}>
        <MUIDataTable
          title={"ACME Employee list"}
          data={data}
          columns={columns}
          options={options}
        />
      </MuiThemeProvider>
    );
  }
}
ReactDOM.render(<App />, document.getElementById("root"));
```

Here is how it looks changed
![stackedFullWidth_fixed](https://user-images.githubusercontent.com/46763360/73977242-dadbd100-48de-11ea-995c-1c704a13acf4.jpg)

## Steps to Reproduce (for bugs)

Example is here
https://codesandbox.io/s/muidatatables-custom-toolbar-0udlv

## Your Environment

| Tech         | Version |
|--------------|---------|
| Material-UI  |   4.9.1      |
| MUI-datatables  |  2.14.0       |
| React        |  16.12.0       |
| browser      |   Chromium  Version 71.0.3578.98  / Firefox Quantum      |
| etc          |         |

I hope it helps, thank you.

**Comment by alainib** [comment]
nice solution. thank i have the same problem.
is there a way to have title and content on same line :
like this : 

**Name**   Gabby George
**Title**     Business Analyst

and not like this 

**Name**   
Gabby George
**Title**     
Business Analyst

thanks

**Comment by patorjk** [comment]
@ramosmarco I've recently taken over as maintainer of this table (I've been working on it for a while and took over after the other two didn't have time anymore) and noticed this as well when I was checking out the stacked view (looks like it was introduced in v2.14.0). I really like what you've come up with as a solution. If you want to put in a PR to the v3 branch I'll make sure this gets into the v3 release.

**Comment by ramosmarco** [comment]
Thank you Patorjk. Honestly I have very little css knowledge, I found the solution but I’m not sure if it’s the best way to do it, or if it’s going to work on every browser, so I think it’s better if someone more experienced review it.

Cheers

**Comment by patorjk** [comment]
@ramosmarco No worries, thank you for the reply. I've put up an issue for community feedback to see if anyone has any strong feelings - https://github.com/gregnb/mui-datatables/issues/1308

I will go ahead and tackle this. I may use your design or I may see if I can fix  the current solution to work for everyone's use cases. If I do the latter, I may also add another display option for your styling, but I don't want to get too far ahead of myself.

**Comment by patorjk** [comment]
Issues with the stacked/vertical mode are fixed in version 3.0.0 of the table. A new "simple" responsive mode was also introduced.
